### ISSUE TITLE
Consider supporting RabbitMQ custom "x-" headers

### ISSUE THREAD

**Issue by andrewlorenz** [rabbitmq/rabbitmq-server#2591]
When sending a message to an exchange defined of type "x-delayed-message" (via the RabbitMQ plugin), there is a need to set a header value for "x-delay". The documented method of setting headers is by passing an "application_properties" object to the .send method, but this appears to not make it to the message as sent to the exchange.  My test code is:
```container.on('sendable', function (context) {
  context.sender.send({
    message_id: 'test-id-1',
    subject: 'tester',
    application_properties: {
      'x-delay': 30000,
    },
    body: {
      sequence: 'test-body-1',
    },
  });
});
```

This message is delivered fine, but it arrives immediately at the queue bound to the exchange.
The README for this repo states rather cryptically that

"_currently we discard application headers (of AMQP 1.0 messages) and headers (of AMQP 0-9-1 messages sent through to AMQP 1.0). In other words, the (AMQP 1.0) application headers section is only available to AMQP 1.0 clients, and the (AMQP 0-9-1) headers field is only available to AMQP 0-9-1 clients._"

This statement actually implies a contradiction in that the application headers (which I am assuming actually means the 'application properties') are both discarded for AMQP 1.0, and only available to AMQP 1.0 clients!

Unfortunately none of the sample code snippets show any use of AMQP headers, so its not possible to work out how to set them, or even if the headers are actually supported.

**Comment by AlexandruChiritaOrpak** [comment]
I know we might not be talking the same thing, but the title is good for what I'm commenting also.

If you want to have the message_id value even after a message goes to the dead letter queue, you can have it like `x-message-id` as a special header that contains the message id because it is kept even after NACK.

I tested it and it worked for me with RabbitMQ.Client library in C# and amqp091-go for GO (messages NACKed keep their message id this way).

So right now you can publish a message with `x-message-id` that is being kept even in the dead letter queue so that you can use if you want to trace a message by it's id even in the dead letter queue.

**Comment by ansd** [comment]
Fixed in RabbitMQ 4.0
